### DISCUSSION TITLE
Add Banner Support for generated USDZ

### DISCUSSION THREAD

**Discussion by GianiNavette** [google/model-viewer#3811] [Q&A] [answered]
### Description
Hello, I want to be able to add a banner to my model but without having to generate and usdz file to put inside the link

As of right now 
you can put it inside the attribute "ios-src"

`ios-src="https://example.com/biplane.usdz#custom=https://example.com/customBanners/comingSoonBanner.html"`

What I am suggesting is adding an attribute called
"ios-banner" where you only put the link of the banner
and in the src attribute you put your GLB file
like so

`src="https://example.com/biplane.glb" ios-banner="https://example.com/customBanners/comingSoonBanner.html"`

### Version
 model-viewer: Last Version

### Browser Affected
- [X] Safari

### OS

- [X] iOS

### AR
- [x] QuickLook

**Comment by elalish** [comment, marked as answer]
Actually this should already work, but we should really document it better. Try `src="https://example.com/biplane.glb#custom=https://example.com/customBanners/comingSoonBanner.html"`

This should work for both the Quick Look and Scene Viewer API (they're non-intersecting, so you can safely use both in the same URL).

**Reply by GianiNavette** [reply]
> Actually this should already work, but we should really document it better. Try `src="https://example.com/biplane.glb#custom=https://example.com/customBanners/comingSoonBanner.html"`
> 
> This should work for both the Quick Look and Scene Viewer API (they're non-intersecting, so you can safely use both in the same URL).

Okay didn't know it was possible, thank you very much

**Reply by milesgreen** [reply]
I think you'll be needing ModelViewer v2.0 to take advantage of it.

**Reply by elizaII** [reply]
Hi! I added the [URL parameters](https://developer.apple.com/documentation/arkit/adding_an_apple_pay_button_or_a_custom_action_in_ar_quick_look) at the end of `src` to display the banner in Quick Look for the auto-generated .usdz-file (it works perfect). However, that made SceneViewer on my Android throw an error message saying: "Couldn't load object" (similar to [the last comment](https://github.com/google/model-viewer/discussions/3632#discussioncomment-3284644) in #3632 ). 

If I remove the parameters, the model loads fine in SceneViewer.

How can I display a banner for the auto-generated usdz-file while also being able to load the model in SceneViewer?

**Reply by elalish** [reply]
I'd recommend WebXR over SceneViewer, as it's faster and more feature-rich. Does that solve the problem?

**Comment by btutal** [comment]
Is there any best practice for the usage of banners in WebXR mode on Android and Quick Look on iOS together?

**Reply by elalish** [reply]
Unfortunately it's all separate; For WebXR you use HTML child elements, for Quick Look you use parameters appended to the `src` URL. We should probably make an example.

**Reply by jdosses** [reply]
Does the comingsoonbanner.html only works for scene-viewer and QuickLook? I've tried the html child elements with ar-status with no success, are there any examples? Thank you.
Quicklook is so much easier to deal with 🫤

**Reply by elalish** [reply]
https://modelviewer.dev/examples/augmentedreality/index.html#webXR
